trieste/models/optimizer.py

```diff
@@ -41,27 +46,31 @@
 
 OptimizeResult = Union[scipy.optimize.OptimizeResult, None]
 """
-Optimization result. For scipy optimizer it is :class:`~scipy.optimize.OptimizeResult`.
-TensorFlow optimizer doesn't return any result.
+Optimization result. TensorFlow optimizer doesn't return any result. For scipy optimizer that is
+also commonly used, it is :class:`~scipy.optimize.OptimizeResult`.
 """
 
 
 @dataclass
 class Optimizer:
     """Optimizer for training models with all the training data at once."""
 
-    optimizer: gpflow.optimizers.Scipy | tf.optimizers.Optimizer
-    """ The underlying optimizer to use. """
+    optimizer: Any
```

Comment:
on aside, why GPflow has its own scipy optimizer, is it any different than the actual scipy one?

Reply:
There are a lot more types of optimizers that are not covered here, e.g. MCMC, Particle filters, evolutionary algorithms. I'm not saying you would use these with say GPs, but as a user you might implement an additional model which needs a different optimizer. My thinking was that `Any` then allows passing whatever optimizer, rather than subclassing `Optimizer` and setting the optimizer to your type. I also tried to contain usage of gpflow within `models/gpflow` sub-package, it made sense to me, but perhaps there is not a strong argument for that.